docs/reference/query-dsl/multi-term-rewrite.asciidoc

```diff
@@ -1,45 +1,79 @@
-multi term queries and end up going through a process of rewrite. This
-also happens on the
-<<query-dsl-query-string-query,query_string>>.
-All of those queries allow to control how they will get rewritten using
-the `rewrite` parameter:
-
-* `constant_score` (default): A rewrite method that performs like
-`constant_score_boolean` when there are few matching terms and otherwise
-visits all matching terms in sequence and marks documents for that term.
-Matching documents are assigned a constant score equal to the query's
-boost.
-* `scoring_boolean`: A rewrite method that first translates each term
-into a should clause in a boolean query, and keeps the scores as
-computed by the query. Note that typically such scores are meaningless
-to the user, and require non-trivial CPU to compute, so it's almost
-always better to use `constant_score`. This rewrite method will hit
-too many clauses failure if it exceeds the boolean query limit (defaults
-to `1024`).
-* `constant_score_boolean`: Similar to `scoring_boolean` except scores
-are not computed. Instead, each matching document receives a constant
-score equal to the query's boost. This rewrite method will hit too many
-clauses failure if it exceeds the boolean query limit (defaults to
-`1024`).
-* `top_terms_N`: A rewrite method that first translates each term into
-should clause in boolean query, and keeps the scores as computed by the
-query. This rewrite method only uses the top scoring terms so it will
-not overflow boolean max clause count. The `N` controls the size of the
-top scoring terms to use.
-* `top_terms_boost_N`: A rewrite method that first translates each term
-into should clause in boolean query, but the scores are only computed as
-the boost. This rewrite method only uses the top scoring terms so it
-will not overflow the boolean max clause count. The `N` controls the
-size of the top scoring terms to use.
-* `top_terms_blended_freqs_N`: A rewrite method that first translates each
-term into should clause in boolean query, but all term queries compute scores
-as if they had the same frequency. In practice the frequency which is used
-is the maximum frequency of all matching terms. This rewrite method only uses
-the top scoring terms so it will not overflow boolean max clause count. The
-`N` controls the size of the top scoring terms to use.
+== `rewrite` Parameter
+
+WARNING: This parameter is for expert users only. Changing the value of this
+parameter can impact search performance and relevance.
+
+To improve performance, {es} rewrites the following queries:
+
+* <<query-dsl-prefix-query, `prefix`>>
+* <<query-dsl-query-string-query, `query_string`>>
+* <<query-dsl-wildcard-query, `wildcard`>>
+
+In their original form, these queries are expensive. {es} changes the
+original query to a less expensive <<query-dsl-bool-query, `bool` query>>. This
+`bool` query contains a `should` clause and <<query-dsl-term-query, `term`
+query>> for each matching term.
```

Comment:
I am not sure this is always the case, but may be @jimczi  can confirm here. As I understood it sometimes for `constant_score` instead of a boolean query,  we just use a document bit set.  And it looks liek the rewrite is done by Lucene, not ES